packages/browser/tests/redux/backtraceReduxMiddlewareTests.spec.ts

```diff
@@ -149,20 +178,59 @@ describe('createBacktraceReduxMiddleware', () => {
                 payload,
             };
-            const store = getStore((action: Action) => {
-                if (action.type === 'test/addToTestArray') {
-                    return {
-                        ...action,
-                        type: expectedType,
-                    };
-                } else {
-                    throw new Error("action.type 'test/addToTestArray' never called");
-                }
+            const store = getStore({
+                interceptAction: (action: Action) => {
+                    if (action.type === 'test/addToTestArray') {
+                        return {
+                            ...action,
+                            type: expectedType,
+                        };
+                    } else {
+                        throw new Error("action.type 'test/addToTestArray' never called");
+                    }
+                },
             });
             const breadcrumbsSpy = getBreadcrumbsSpy('info');
             const expected = getExpectedBreadcrumb(expectedAction);
             store.dispatch(addToTestArray(payload));
-            expect(breadcrumbsSpy).toHaveBeenCalledWith(expected);
+            expect(breadcrumbsSpy).toHaveBeenCalledWith(...expected);
+        });
+
+        describe('modes', () => {
+            it('Should add only action type as JSON to breadcrumb attributes by default', () => {
+                const interceptedAction = { type: 'expected-type', payload: { abc: '123' } };
+                const store = getStore({ interceptAction: () => interceptedAction });
+                const breadcrumbsSpy = getBreadcrumbsSpy('info');
+                const expected = getExpectedBreadcrumb(interceptedAction, 'omit-values');
+                store.dispatch(addToTestArray('Message to add'));
+                expect(breadcrumbsSpy).toHaveBeenCalledWith(...expected);
+            });
+
+            it('Should add whole action as JSON to breadcrumb attributes when mode=all', () => {
+                const interceptedAction = { type: 'expected-type', payload: { abc: '123' } };
+                const store = getStore({ interceptAction: () => interceptedAction, mode: 'all' });
+                const breadcrumbsSpy = getBreadcrumbsSpy('info');
+                const expected = getExpectedBreadcrumb(interceptedAction, 'all');
+                store.dispatch(addToTestArray('Message to add'));
+                expect(breadcrumbsSpy).toHaveBeenCalledWith(...expected);
+            });
+
+            it('Should add only action type as JSON to breadcrumb attributes when mode=omit-values', () => {
+                const interceptedAction = { type: 'expected-type', payload: { abc: '123' } };
+                const store = getStore({ interceptAction: () => interceptedAction, mode: 'omit-values' });
+                const breadcrumbsSpy = getBreadcrumbsSpy('info');
+                const expected = getExpectedBreadcrumb({ type: interceptedAction.type }, 'omit-values');
```

Comment:
shouldn't you pass the whole interceptedAction here?

Reply:
updated